### ISSUE TITLE
Incremental build of builders fail

### ISSUE THREAD

**Issue by vukisoft** [google/FreeBuilder#111]
I realized that in some cases freebuilder fails to work unless I clean first.
In the following example, if I add the first only, it compiles (and works in IDE).  However if I add the second too, all fails. It gets successful again if I do clean build (or clean in IDE - Eclipse).
This leads us to a kind of undeterministic behaviour, sometimes it works sometime it doesn't (until I clean).

```
public class Main {
  interface Foo<T> {
    T getBar();
  }

  @FreeBuilder
  interface GenericFoo<T> extends Foo<T> {
    class Builder<T> extends Main_GenericFoo_Builder<T> {}
  }

  @FreeBuilder
  interface StringFoo extends Foo<String> {
    class Builder extends Main_StringFoo_Builder {}
  }
}
```

We also found a situation where it does not work in IDE only, but otherwise builds properly, but I couldn't create an example class of that yet.

**Comment by cjp39** [comment]
Could you attach the errors you're seeing?

**Comment by cjp39** [comment]
What is "T" in interface Foo?

**Comment by cjp39** [comment]
Okay, here are the errors I'm seeing when I do this in Eclipse 4.5.0:

- Two "Source file already created" warnings, one on each `@FreeBuilder` annotation
- An error on the `Builder` class in `GenericFoo<T>`, "Builder extends the wrong type (should be Main_GenericFoo_Builder)", which is generated by FreeBuilder and looks erroneous
- A placeholder type is generated for `Main_GenericFoo_Builder<T>`
- Multiple errors in `Main_StringFoo_Builder` caused by FreeBuilder generating invalid code that references undefined type `T` instead of `String`

When I clean and regenerate, I still see the errors in `Main_StringFoo_Builder`.

Does this match what you're seeing?

**Comment by vukisoft** [comment]
For incremental build I see
Main.java:[23,27] cannot find symbol
symbol:   class Main_StringFoo_Builder
location: interface foo.bar.Main.StringFoo
Main.java:[17,5] Builder extends the wrong type (should be Main_GenericFoo_Builder)

I don't see errors for clean build.
Also I don't see any of the warnings, but probably you're correct :)

So you will keep this and add separate bug, too?

**Comment by cjp39** [comment]
Yes, though honestly this sounds like an eclipse bug, which is unlikely to have a workaround we can implement

**Comment by cjp39** [comment]
Opened #112 for the other bug. Thanks!

**Comment by vukisoft** [comment]
The error I quoted comes when building by maven! So it had nothing to do with Eclipse.
So:
mvn install with first inner class - ok
mvn install with both - fail
mvn clean install with both - ok

**Comment by cjp39** [comment]
That's both really weird and really helpful. Thanks!

**Comment by j-baker** [comment]
Also been seeing this.

**Comment by chrisalice** [comment]
This should be fixed in [FreeBuilder 1.7.1](https://github.com/google/FreeBuilder/releases/tag/v1.7.1). Sorry for the delay!
